src/modules/0.0.19/aws_account/index.ts

```diff
@@ -36,35 +38,122 @@ class AccountMapper extends MapperBase<AwsAccountEntity> {
   }
 }
 
+class RegionsMapper extends MapperBase<AwsRegions> {
+  module: AwsAccount;
+  entity = AwsRegions;
+  equals = (a: AwsRegions, b: AwsRegions) => a.region === b.region; // Checking only the region eliminates db/cloud `update` calls
+
+  getRegions = crudBuilderFormat<EC2, 'describeRegions', string[]>(
+    'describeRegions',
+    () => ({}),
+    res => res?.Regions?.map(r => r.RegionName ?? '').filter(r => r !== '') ?? [],
+  );
+
+  cloud = new Crud2<AwsRegions>({
+    create: async (e: AwsRegions[], ctx: Context) => {
+      // Just immediately revert, we can't create regions in the cloud
+      const out = await this.module.awsRegions.db.delete(e, ctx);
+      if (!out || out instanceof Array) return out;
+      return [out];
+    },
+    read: async (ctx: Context, region?: string) => {
+      let client;
+      try {
+        client = await ctx.getAwsClient();
+      } catch (_) {
+        // The initial install of the module will fail to generate a valid client, so just
+        // catch this and pretend nothing bad happened.
+        return [];
+      }
+      if (region) {
+        const awsRegion = new AwsRegions();
+        awsRegion.isDefault = false;
+        awsRegion.isEnabled = true;
+        awsRegion.region = region;
+        return awsRegion;
+      }
+      return (await this.getRegions(client.ec2client)).map(r => {
+        const awsRegion = new AwsRegions();
+        awsRegion.isDefault = false;
+        awsRegion.isEnabled = true;
+        awsRegion.region = r;
+        return awsRegion;
+      });
+    },
+    update: async (_e: AwsRegions[], _ctx: Context) => {
```

Comment:
in the create/update, could we manage/restrict somehow just having a single default region? and how do we manage the flow in case that default region is not enabled?

Reply:
So right now the code has a final fallback to `us-east-1` if the default isn't set.